ComputeEffAccWeightedAvg.py

```diff
@@ -0,0 +1,72 @@
+'''
+Script for the combination of efficiency and acceptance factors
+run: python ComputeEffAccWeightedAvg.py effFileAll.root effFileNonRes.root effFileKStar.root effFileDelta.root effFileLambda1520.root outFileName.root
+'''
+
+import argparse
+import numpy as np
+from ROOT import gROOT, TFile, TCanvas, TLegend  # pylint: disable=import-error,no-name-in-module
+from ROOT import kBlack, kFullDiamond, kRed, kAzure, kFullCircle, kOpenSquare  # pylint: disable=import-error,no-name-in-module
+from utils.StyleFormatter import SetGlobalStyle, SetObjectStyle
+
+parser = argparse.ArgumentParser(description='Arguments')
+parser.add_argument('effFileAll', metavar='text', default='')
+parser.add_argument('effFileNonRes', metavar='text', default='')
+parser.add_argument('effFileKStar', metavar='text', default='')
+parser.add_argument('effFileDelta', metavar='text', default='')
+parser.add_argument('effFileLambda1520', metavar='text', default='')
+parser.add_argument('outFileName', metavar='text', default='')
+parser.add_argument("--batch", help="suppress video output", action="store_true")
+args = parser.parse_args()
+
+gROOT.SetBatch(args.batch)
+SetGlobalStyle(padleftmargin=0.14, padbottommargin=0.12, titlesize=0.045, labelsize=0.04)
+
+effFileAll = TFile.Open(args.effFileAll)
+hEffAllPrompt = effFileAll.Get('hAccEffPrompt')
+hEffAllFD = effFileAll.Get('hAccEffFD')
+SetObjectStyle(hEffAllPrompt, color=kRed+1, markerstyle=kFullCircle)
+SetObjectStyle(hEffAllFD, color=kAzure+4, markerstyle=kOpenSquare, markersize=1.5, linewidh=2, linestyle=7)
+
+effFileNonRes = TFile.Open(args.effFileNonRes)
+hEffNonResPrompt = effFileNonRes.Get('hAccEffPrompt')
+hEffNonResFD = effFileNonRes.Get('hAccEffFD')
+SetObjectStyle(hEffNonResPrompt, color=kRed+1, markerstyle=kFullCircle)
+SetObjectStyle(hEffNonResFD, color=kAzure+4, markerstyle=kOpenSquare, markersize=1.5, linewidh=2, linestyle=7)
+
+effFileKStar = TFile.Open(args.effFileKStar)
+hEffKStarPrompt = effFileKStar.Get('hAccEffPrompt')
+hEffKStarFD = effFileKStar.Get('hAccEffFD')
+SetObjectStyle(hEffKStarPrompt, color=kRed+1, markerstyle=kFullCircle)
+SetObjectStyle(hEffKStarFD, color=kAzure+4, markerstyle=kOpenSquare, markersize=1.5, linewidh=2, linestyle=7)
+
+effFileDelta = TFile.Open(args.effFileDelta)
+hEffDeltaPrompt = effFileDelta.Get('hAccEffPrompt')
+hEffDeltaFD = effFileDelta.Get('hAccEffFD')
+SetObjectStyle(hEffDeltaPrompt, color=kRed+1, markerstyle=kFullCircle)
+SetObjectStyle(hEffDeltaFD, color=kAzure+4, markerstyle=kOpenSquare, markersize=1.5, linewidh=2, linestyle=7)
+
+effFileLambda1520 = TFile.Open(args.effFileLambda1520)
+hEffLambda1520Prompt = effFileLambda1520.Get('hAccEffPrompt')
+hEffLambda1520FD = effFileLambda1520.Get('hAccEffFD')
+SetObjectStyle(hEffLambda1520Prompt, color=kRed+1, markerstyle=kFullCircle)
+SetObjectStyle(hEffLambda1520FD, color=kAzure+4, markerstyle=kOpenSquare, markersize=1.5, linewidh=2, linestyle=7)
```

Comment:
I re-wrote everything using loops in python. It took me sometime to get everything working since I am not super-expert of python. Nevertheless, I checked the output and the result is identical to the old version of the script.